nep-11.mediawiki

```diff
@@ -0,0 +1,418 @@
+<pre>
+  NEP: 11
+  Title: Non-Fungible Token Standard
+  Author: Joe Stewart <hal0x2328@splyse.tech>, Shane Mann <fetter@splyse.tech>, Wyatt Mufson <wyatt@ryu.games>, Erik Zhang <erik@neo.org>
+  Type: Standard
+  Status: Accepted
+  Created: 2018-04-18
+</pre>
+
+==Abstract==
+
+This NEP defines a standard non-fungible token system for the NEO Smart Economy.
+
+==Motivation==
+
+NFTs are required to track, exchange and enforce ownership of digital assets.
+A non-fungible token (NFT) can be thought of like a property deed - each one is unique and carries some non-mutable information (e.g. the physical address of the property).
+Other information, such as the owner of the property, can be changed.
+Also, we provide built-in optional divisibility within each non-fungible asset.
+This allows for high value objects to be tokenized more effectively.
+
+==Specification==
+
+===Common methods===
+
+====symbol====
+
+<pre>
+{
+  "name": "symbol",
+  "parameters": [],
+  "returntype": "String"
+}
+</pre>
+
+Returns a short string symbol of the token managed in this contract. e.g. <code>"MNFT"</code>.
+This symbol SHOULD be short (3-8 characters is recommended), with no whitespace characters or new-lines and SHOULD be limited to the uppercase latin alphabet (i.e. the 26 letters used in English).
+
+This method MUST always return the same value every time it is invoked.
+
+====decimals====
+
+<pre>
+{
+  "name": "decimals",
+  "parameters": [],
+  "returntype": "Integer"
+}
+</pre>
+
+Returns the number of decimals used by the token - e.g. <code>8</code>, means to divide the token amount by <code>100,000,000</code> to get its user representation.
+
+If the token managed in this contract is indivisible, the function SHOULD return <code>0</code>.
+
+If this method returns <code>0</code>, the "Non-divisible NFT methods" must be implemented.
+Otherwise, the "Divisible NFT methods" must be implemented.
+
+This method MUST always return the same value every time it is invoked.
+
+====totalSupply====
+
+<pre>
+{
+  "name": "totalSupply",
+  "parameters": [],
+  "returntype": "Integer"
+}
+</pre>
+
+Returns the total token supply currently in the system.
+
+====balanceOf====
+
+<pre>
+{
+  "name": "balanceOf",
+  "parameters": [
+    {
+      "name": "owner",
+      "type": "Hash160"
+    }
+  ],
+  "returntype": "Integer"
+}
+</pre>
+
+Returns the total amount of NFTs owned by the specified address.
```

Comment:
What is the supposed return value of this method for divisible NFTs? They have an overloaded `balanceOf` which is quite obvious, but they at the same time have this one also (with `owner` parameter only) and if some owner has 0.5 of token A and 0.125 of token B then what is the total amount of divisible NFTs owned?

Reply:
`0.625`

Reply:
But does it have any meaning? These tokens are non-fungible and it can be like you own 0.5 of some house and 0.125 of the other, you don't own 0.625 of anything.